src/auxinfo/participant.rs

```diff
@@ -511,20 +510,14 @@ impl AuxInfoParticipant {
 #[cfg_attr(feature = "flame_it", flame("auxinfo"))]
 fn new_auxinfo<R: RngCore + CryptoRng>(
     rng: &mut R,
-    _prime_bits: usize,
 ) -> Result<(AuxInfoPrivate, AuxInfoPublic, AuxInfoWitnesses)> {
-    // Pull in pre-generated safe primes from text file (not a safe operation!).
-    // This is meant to save on the time needed to generate these primes, but
-    // should not be done in a production environment!
-    let safe_primes = crate::utils::get_safe_primes();
-    let two_safe_primes = safe_primes.iter().choose_multiple(rng, 2);
-
-    // FIXME: do proper safe prime generation
-    //let p = BigNumber::safe_prime(prime_bits);
-    //let q = BigNumber::safe_prime(prime_bits);
-
-    let p = two_safe_primes[0].clone();
-    let q = two_safe_primes[1].clone();
+    #[cfg(not(test))]
+    let (p, q) = (
+        crate::utils::get_random_safe_prime_512(rng),
+        crate::utils::get_random_safe_prime_512(rng),
+    );
+    #[cfg(test)]
+    let (p, q) = crate::utils::get_prime_pair_from_pool_insecure(rng);
```

Comment:
nit: Can you add a little inline documentation here explaining this choice for our future selves? Maybe mention the timing discrepancy.